### PR TITLE
[feat][sale widget] Move CC warning drawer to the root

### PR DESCRIPTION
**Pull request by mimi-imtbl**
### Hi👋, please prefix this PR's title with:

- [ ] `breaking-change:` if you have introduced modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities.
- [x] `feat:`, `fix:`, `refactor:`, `docs:`, or `chore:`.

## Changed

Move the CC warning component to the root so it shows up globally.

### CODE REVIEW

File: packages/checkout/widgets-lib/src/widgets/sale/SaleWidget.tsx [hunk 1]
```diff
@@ -183,6 +187,10 @@ export default function SaleWidget(props: SaleWidgetProps) {
               subheading={viewState.view.data?.subheading}
             />
           )}
+          <CreditCardWarningDrawer
+            visible={showCreditCardWarning}
```
Comment (dreamoftrees) [inline packages/checkout/widgets-lib/src/widgets/sale/SaleWidget.tsx:191]: I was wondering how does this get triggered now? After the move I can't see how it links up from the payment methods 🤔

**Review by dreamoftrees — COMMENTED** [review]
@mimi-imtbl just a little unsure after this change under which conditions the credit card warning is shown?

File: packages/checkout/widgets-lib/src/widgets/sale/SaleWidget.tsx (hunk 1 above)
Comment (mimi-imtbl) [reply packages/checkout/widgets-lib/src/widgets/sale/SaleWidget.tsx:193]: accidentally removed in the latest `Clean up` commit, reverted it back

File: packages/checkout/widgets-lib/src/widgets/sale/components/CreditCardWarningDrawer.tsx [hunk 2]
```diff
@@ -4,19 +4,24 @@ import {
...
 export function CreditCardWarningDrawer({
   visible,
   setShowCreditCardWarning,
-  setPaymentMethod,
 }: CreditCardWarningDrawerProps) {
   const { t } = useTranslation();
+  const { setPaymentMethod } = useSaleContext();
+
+  const handleCtaButtonClick = () => {
+    setShowCreditCardWarning(false);
```
Comment (jhesgodi) [inline packages/checkout/widgets-lib/src/widgets/sale/components/CreditCardWarningDrawer.tsx:22]: shouldn't be `true`?

**Review by jhesgodi — APPROVED** [review]

**Review by dreamoftrees — APPROVED** [review]
👍 Nice one @mimi-imtbl

**Pull request merged**